From 1b9128bae737fa559f5e2c191d6679a856efbad9 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Wed, 23 Feb 2011 22:23:25 +0800
Subject: [PATCH 10/32] __xmlRaiseError: remove redundant schannel
 initialization

In case the domain is XML_FROM_VALID, ctxt and schannel are already initialized
earlier with the same data
---
 error.c |    8 --------
 1 file changed, 8 deletions(-)

diff --git a/error.c b/error.c
index 34472eb..15b431e 100644
--- a/error.c
+++ b/error.c
@@ -475,14 +475,6 @@ __xmlRaiseError(xmlStructuredErrorFunc schannel,
 	if (schannel != NULL)
 	    data = xmlStructuredErrorContext;
     }
-    if ((domain == XML_FROM_VALID) &&
-        ((channel == xmlParserValidityError) ||
-	 (channel == xmlParserValidityWarning))) {
-	ctxt = (xmlParserCtxtPtr) ctx;
-	if ((schannel == NULL) && (ctxt != NULL) && (ctxt->sax != NULL) &&
-	    (ctxt->sax->initialized == XML_SAX2_MAGIC))
-	    schannel = ctxt->sax->serror;
-    }
     /*
      * Formatting the message
      */
-- 
1.7.9.5

